drivers/flash/flash_mspi_nor.c

```diff
@@ -450,53 +619,39 @@ static int flash_chip_init(const struct device *dev)
 		return rc;
 	}
 
-	dev_data->xfer.xfer_mode  = MSPI_PIO;
-	dev_data->xfer.packets    = &dev_data->packet;
-	dev_data->xfer.num_packet = 1;
-	dev_data->xfer.timeout    = 10;
-
-	dev_data->xfer.cmd_length  = 1;
-	dev_data->xfer.addr_length = 0;
-	dev_data->xfer.tx_dummy    = 0;
-	dev_data->xfer.rx_dummy    = 0;
-
-	dev_data->packet.dir       = MSPI_RX;
-	dev_data->packet.cmd       = JESD216_CMD_READ_ID;
+	command_set(dev, &commands[MSPI_IO_MODE_SINGLE].id);
 	dev_data->packet.data_buf  = id;
 	dev_data->packet.num_bytes = sizeof(id);
+
 	rc = mspi_transceive(dev_config->bus, &dev_config->mspi_id,
 			     &dev_data->xfer);
 	if (rc < 0) {
-		LOG_ERR("Failed to read JEDEC ID in single line mode: %d", rc);
+		LOG_ERR("Failed to read JEDEC ID in initial line mode: %d", rc);
 		return rc;
 	}
 
-	/*
+	/* For octal mode:
 	 * If the read ID does not match the one from DTS, assume the flash
 	 * is already in the Octa I/O mode, so switching it is not needed.
 	 */
```

Comment:
Maybe there should be a note about QUAD mode too. Right now it is easy to overlook that for QUAD this if is true.